sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+
+from ..utils.validation import check_is_fitted
+from ..utils.validation import column_or_1d
+from ._encoders import _BaseEncoder
+
+
+class TargetRegressorEncoder(_BaseEncoder):
+    """Target Encoder for Regression Targets.
+
+    Each category is encoded based its effect on the target variable. The
+    encoding scheme takes a weighted average estimated by a multilevel
+    linear model.
+
+    Read more in the :ref:`User Guide <target_regressor_encoder>`.
+
+    Parameters
+    ----------
+    categories : 'auto' or a list of array-like, default='auto'
+        Categories (unique values) per feature:
+
+        - 'auto' : Determine categories automatically from the training data.
+        - list : `categories[i]` holds the categories expected in the ith
+          column. The passed categories should not mix strings and numeric
+          values within a single feature, and should be sorted in case of
+          numeric values.
+
+        The used categories can be found in the `categories_` attribute.
+
+    Attributes
+    ----------
+    cat_encodings_ : list of ndarray
+        The encoding corresponding to the categories in the `categories_`
+        attribute.
```

Comment:
Should this also have "
        (in order of the features in X and corresponding with the output
        of :meth:`transform`). "... ?